packages/@tinacms/toolkit/src/packages/form-builder/FormBuilder.tsx

```diff
@@ -173,7 +180,12 @@ export const FormBuilder: FC<FormBuilderProps> = ({
                     )}
                     <Button
                       onClick={() => handleSubmit()}
-                      disabled={pristine || submitting || invalid}
+                      disabled={
+                        pristine ||
+                        submitting ||
+                        hasValidationErrors ||
+                        (invalid && !dirtySinceLastSubmit)
```

Comment:
From the loom, I think `hasValidationErrors` should catch the live-form validation issues. 
As is, `invalid` will always be false once a submission error has occurred (since it locks the form from being submitted again): https://final-form.org/docs/final-form/types/FieldState#invalid

Reply:
@Phoenix-Alpha Sorry I missed this message but @jamespohalloran explained it well. 
After a failed submission `invalid` stays true, so there was no way to re-enable the button without resetting the entire form, losing all the values entered.

This way the button is always disabled when there are errors on the fields, but will be enabled after a failed submission if all fields are valid and there have been changes to at least one of the fields.